### PR TITLE
AWS target for robocup

### PR DESCRIPTION
**Pull request by Y0sh1dk**
`./b webots build webotsrobocup --target g4dnxlarge`

### CODE REVIEW

File: docker/usr/local/toolchain/generate_g4dnxlarge_toolchain.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#! /usr/bin/env python3
```
Comment (KipHamiltons) [inline docker/usr/local/toolchain/generate_g4dnxlarge_toolchain.py:1]: I would love a comment at the top of this explaining what this target is for. Future years will thank us

File: tools/webots.py [hunk 2]
```diff
@@ -178,9 +198,9 @@ def exec_push():
         sys.exit(exit_code)
 
 
-def run(sub_command, roles=None, role=None, **kwargs):
+def run(sub_command, roles=None, role=None, target="generic", **kwargs):
```
Comment (KipHamiltons) [inline tools/webots.py:201]: IMO the default should be this. AMD and old intel users can use the flag lel
```suggestion
def run(sub_command, roles=None, role=None, target="g4dnxlarge", **kwargs):
```

Comment (KipHamiltons) [reply tools/webots.py:201]: @JosephusPaye what do you reckon

Comment (Y0sh1dk) [reply tools/webots.py:201]: I think It should default to "generic" because we will likely only be running on aws for this robocup

Comment (KipHamiltons) [reply tools/webots.py:201]: we're also only planning to use this webots build tool for this robocup?

Comment (Y0sh1dk) [reply tools/webots.py:201]: I believe the webots tool will still be useful after robocup.

File: docker/usr/local/package/openblas/g4dnxlarge.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
```
Comment (abihall) [inline docker/usr/local/package/openblas/g4dnxlarge.sh:2]: Please add a comment here, I have no idea what i'm looking at

File: tools/webots.py (hunk 2 above)
Comment (CMurtagh-LGTM) [reply tools/webots.py:201]: Set the default to `g4dnxlarge`, if people want to build generic they can do a normal build.

As this module stand it makes sense to have this target, if we want to change it in the furture, we can change the default then.

File: tools/webots.py [hunk 4]
```diff
@@ -78,6 +92,12 @@ def exec_build(roles):
         cprint(f"unable to build code, exit code {exit_code}", "red", attrs=["bold"])
         sys.exit(exit_code)
 
+    print("Setting target 'generic'...")
```
Comment (abihall) [inline tools/webots.py:95]: Why do you set the target to the passed target and then set the target to generic after clean, configure and build? A comment would be nice here if possible. Comments a nice here for people that don't understand... like me

File: tools/webots.py [hunk 5]
```diff
@@ -178,9 +198,9 @@ def exec_push():
...
 
 
-def run(sub_command, roles=None, role=None, **kwargs):
+def run(sub_command, roles=None, role=None, target="generic", **kwargs):
     if sub_command == "build":
-        exec_build(roles)
+        exec_build(target, roles)
     elif sub_command == "push":
         exec_push()
     elif sub_command == "run":  # For testing
```
Comment (abihall) [inline tools/webots.py:206]: What is this comment? Not very clear what you are trying to do here

**Review by abihall — COMMENTED** [review]
A few comment requests, I cannot understand your code without some comments on what the code is achieving.

File: tools/webots.py (hunk 5 above)
Comment (Y0sh1dk) [reply tools/webots.py:206]: The image built is intended to be ran on AWS for RoboCup, the `./b webots run` is usually used just for testing the built image.

File: tools/webots.py (hunk 4 above)
Comment (Y0sh1dk) [reply tools/webots.py:95]: 99% of the time we are compiling for target `generic`. So its probably best to set it back to it after building a docker image for the given target.

File: tools/webots.py (hunk 2 above)
Comment (Y0sh1dk) [reply tools/webots.py]: Finneee

File: tools/webots.py (hunk 4 above)
Comment (KipHamiltons) [reply tools/webots.py:100]:
```suggestion
    print("Setting back to target 'generic'...")
```

**Review by KipHamiltons — APPROVED** [review]

**Comment by KipHamiltons** [comment]
LGTM - Let's get this merged

**Pull request merged**